catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/dialog/UploadProgressDialogFragment.java

```diff
@@ -128,14 +128,14 @@ public void run() {
 			@Override
 			public void onClick(View v) {
 				startWebViewActivity(SHARE_PROGRAM_URL + ServerCalls.getInstance().getProjectId());
-				dismiss();
+				alertDialog.getOwnerActivity().finish();
```

Comment:
according to Android dev documentation `.getOwnerActivity(..)` can in some cases return `null`. so you could just add a null check and call `dismiss()` if the Activity was `null`.